pkg/spiffe/spiffe.go

```diff
@@ -76,17 +95,104 @@ func GenSpiffeURI(ns, serviceAccount string) (string, error) {
 func MustGenSpiffeURI(ns, serviceAccount string) string {
 	uri, err := GenSpiffeURI(ns, serviceAccount)
 	if err != nil {
-		log.Debug(err.Error())
+		spiffeLog.Debug(err.Error())
 	}
 	return uri
 }
 
 // GenCustomSpiffe returns the  spiffe string that can have a custom structure
 func GenCustomSpiffe(identity string) string {
 	if identity == "" {
-		log.Error("spiffe identity can't be empty")
+		spiffeLog.Error("spiffe identity can't be empty")
 		return ""
 	}
 
 	return URIPrefix + GetTrustDomain() + "/" + identity
 }
+
+// RetrieveSpiffeBundleRootCertsFromStringInput retrieves the trusted CA certificates from a list of SPIFFE bundle endpoints.
+// It can use the system cert pool and the supplied certificates to validate the endpoints.
+// The input endpointTuples should be in the json format of:
+// [
+//		{"trustdomain": "foo", "url": "..."},
+//		{"trustdomain": "bar", "url": "..."}
+// ]
+func RetrieveSpiffeBundleRootCertsFromStringInput(inputString string, extraTrustedCerts []*x509.Certificate) (
```

Comment:
If you want to do it this way (i.e. bind the cert to a trust domain - which is correct ) - maybe return a map, with key == trustDomain and value == cert ? 

And I assume the trust domain verification will check that the actual cert is right, since TLS verification treats all certs in the pool as equal.

Reply:
Correct. That's exactly what I'm trying to do in followup PRs.